File: mobile/src/api/auth.api.ts
import apiClient from './client';

export interface UserRole {
  _id: string;
  name: string;
  slug: string;
}

export interface UserProfile {
  _id: string;
  name: string;
  email: string;
  employeeId?: string;
  role: UserRole | string;
  department?: { _id: string; name: string } | string;
  designation?: string;
  avatar?: string;
  phone?: string;
  isActive?: boolean;
}

export interface LoginResponse {
  success: boolean;
  token?: string;
  accessToken?: string;
  refreshToken?: string;
  user: UserProfile;
}

export const AuthApi = {
  async login(identifier: string, password: string): Promise<LoginResponse> {
    const res = await apiClient.post('/api/auth/login', {
      identifier,
      password,
    });
    const payload = res.data?.data || res.data;
    return {
      success: res.data?.success ?? true,
      token: payload.token || payload.accessToken,
      accessToken: payload.token || payload.accessToken,
      refreshToken: payload.refreshToken,
      user: payload.user,
    };
  },

  async getMe(): Promise<{ success: boolean; user: UserProfile }> {
    const res = await apiClient.get('/api/auth/me');
    const userPayload = res.data?.data || res.data?.user || res.data;
    return {
      success: res.data?.success ?? true,
      user: userPayload,
    };
  },

  async logout(): Promise<void> {
    try {
      await apiClient.post('/api/auth/logout');
    } catch (e) {
      // Ignore network errors during logout
    }
  },
};

File: mobile_flutter/lib/features/auth/data/auth_api.dart
import '../../../core/network/api_client.dart';
import '../../../core/network/api_endpoints.dart';
import '../../../models/user_profile.dart';

class LoginResponse {
  final bool success;
  final String? token;
  final String? refreshToken;
  final UserProfile user;

  LoginResponse({
    required this.success,
    this.token,
    this.refreshToken,
    required this.user,
  });
}

class AuthApi {
  final ApiClient _client = ApiClient();

  Future<LoginResponse> login(String identifier, String password) async {
    final response = await _client.dio.post(
      ApiEndpoints.login,
      data: {
        'identifier': identifier,
        'password': password,
      },
    );

    final data = response.data;
    final payload = data['data'] ?? data;

    return LoginResponse(
      success: data['success'] ?? true,
      token: payload['token'] ?? payload['accessToken'],
      refreshToken: payload['refreshToken'],
      user: UserProfile.fromJson(payload['user'] ?? {}),
    );
  }

  Future<UserProfile> getMe() async {
    final response = await _client.dio.get(ApiEndpoints.me);
    final data = response.data;
    final userPayload = data['data'] ?? data['user'] ?? data;
    return UserProfile.fromJson(userPayload);
  }

  Future<void> logout() async {
    try {
      await _client.dio.post(ApiEndpoints.logout);
    } catch (_) {
      // Ignore network errors during logout
    }
  }
}
